checker/src/main/java/org/checkerframework/checker/optional/OptionalImplVisitor.java

```diff
@@ -247,6 +279,8 @@ public void handleTernaryIsPresentGet(ConditionalExpressionTree tree) {
   private boolean sameExpression(ExpressionTree tree1, ExpressionTree tree2) {
```

Comment:
I agree, but it seems like I would need to introduce a dependency on `org.checkerframework.dataflow.expression` if I made that refactoring; is this desirable?

Reply:
The `checker` package already depends on the `dataflow` package, so I think that is OK.

Reply:
The `TreeUtils` class is not in the `checker` package. It is in the `javacutil` package, which I do not think declares a dependency on the `dataflow.expression` package, where `JavaExpression` is defined.

Reply:
@smillst Your thoughts?

Reply:
This method can't be in TreeUtils because the `javacutil` package should not depend on `dataflow`.  We could move JavaExpression to `javacutil`.  This would also required moving  method that use Nodes, i.e.`JavaExpression#fromNode(...)`, to somewhere else in the dataflow package.

I also don't see a better way to this comparison than what is here.

Reply:
Thank you both for your input.

I've added additional context in the Javadoc for this method explaining why it isn't in `TreeUtils` in addition to a link to this discussion. I think this is adequate for now, but I am happy to revisit this if needed.